pandas/core/indexes/datetimelike.py

```diff
@@ -180,6 +180,9 @@ def map(self, mapper, na_action=None):
         except Exception:
             return self.astype(object).map(mapper)
 
+    def argsort(self, *args, **kwargs) -> np.ndarray:
+        return np.argsort(self._data)
```

Comment:
i think if we want this to be consistent across DTI/TDI/PI this needs to take a `.view("M8[ns]")`

Also should this happen at the DTA/TDA/PA level and get passed through?